modules/bakhtiarov_a_priority_queue/src/priority_queue_app.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Fedotov Kirill
+
+#include "include/priority_queue_app.h"
+#include <sstream>
+#include <cstring>
+#include <exception>
+
+bool PriorityQueueApplication::Validate(int argc, char *argv[]) {
+    int required_argc = 4;
+    if (argc == 1) {
+        Help(argv[0]);
+        return false;
+    } else if (std::strcmp(argv[1], "--help") == 0) {
+        Help(argv[0]);
+        return false;
+    } else if (std::strcmp(argv[2], "push") != 0
+               && std::strcmp(argv[2], "pop") != 0
+               && std::strcmp(argv[2], "front") != 0
+               && std::strcmp(argv[2], "back") != 0) {
+        Help(argv[0], "Operation must be 'push', 'pop', 'front' or 'back'.");
+        return false;
+    } else if (argc < required_argc && std::strcmp(argv[2], "push") == 0) {
+        Help(argv[0], "Incorrect number of arguments for push.");
+        return false;
+    } else {
+        try {
+            int size = std::stoi(argv[1]);
+            if (size < 1) {
+                throw std::invalid_argument
+                ("Queue size must be greater than 0.");
+            }
+            queue = TQueue_insert<int>(size);
+        } catch(const std::exception& e) {
+            Help(argv[0], "Invalid arguments.");
```

Comment:
В данном случае лучше вывести конкретное сообщение об ошибке.
```suggestion
            Help(argv[0], e.what());
```